### PR TITLE
Verify that test ownership doesn't go assigned to "Unknown"

### PR DESCRIPTION
**Pull request by stbenjam**
will error out in CI like this:

```
time="2023-05-24T17:30:45Z" level=warning msg="test moved from \"Etcd\" to \"Unknown\"" Name="[sig-etcd][Serial] etcd [apigroup:config.openshift.io] is able to vertically scale up and down with a single node [Timeout:60m][apigroup:machine.openshift.io] [Suite:openshift/conformance/serial]" Suite=openshift-tests
time="2023-05-24T17:30:45Z" level=fatal msg="Components are not allowed to move to Unknown. Please assign correct ownership." 
```

Example failure: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-eng_ci-test-mapping/19/pull-ci-openshift-eng-ci-test-mapping-main-mapping/1661423377182625792

### CODE REVIEW

File: cmd/ci-test-mapping/mapping_verify.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+		// Look for removed mappings
+		removedMaps := make([]cl, 0)
+		for _, n := range newMap {
+			if n.Component == components.DefaultComponent {
+				locator := cl{n.Name, n.Suite}
+				if old, ok := currentComponents[locator]; ok && old != components.DefaultComponent {
+					removedMaps = append(removedMaps, locator)
+				}
+			}
```
Comment (deads2k) [inline cmd/ci-test-mapping/mapping_verify.go:81]:
```suggestion
			if n.Component != components.DefaultComponent {
			    continue
			}
			locator := cl{n.Name, n.Suite}
			if old, ok := currentComponents[locator]; ok && old != components.DefaultComponent {
				removedMaps = append(removedMaps, locator)
			}
```

preferred golang/kube style is to reduce nesting with early continue/return.

File: cmd/ci-test-mapping/mapping_verify.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		}
+
+		logrus.Infof("verification complete in %+v", time.Since(now))
+		if len(removedMaps) > 0 {
```
Comment (deads2k) [inline cmd/ci-test-mapping/mapping_verify.go:85]: similar comment about nesting depth.  could invert and return early.

File: cmd/ci-test-mapping/mapping_verify.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	var mapping []v1.TestOwnership
+	err = json.Unmarshal(body, &mapping)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal JSON: %w", err)
+	}
+
+	return mapping, nil
+}
+
+func init() {
```
Comment (deads2k) [inline cmd/ci-test-mapping/mapping_verify.go:112]: prefer to have a method that calls and registers similar to https://github.com/openshift/crd-schema-checker/blob/master/cmd/crd-schema-checker/main.go#L40-L41

can be a followup given scope.

Comment (stbenjam) [reply cmd/ci-test-mapping/mapping_verify.go:115]: I like that style better, I'll look at doing it later when I have some time

**Pull request merged**